modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_selector/AssetCategoriesSelector.es.js

```diff
@@ -12,64 +12,77 @@
  * details.
  */
 
-import 'clay-icon';
-import Component from 'metal-component';
-import Soy from 'metal-soy';
-import {Config} from 'metal-state';
+import PropTypes from 'prop-types';
+import React from 'react';
 
-import './AssetVocabularyCategoriesSelector.es';
-import templates from './AssetCategoriesSelector.soy';
+import AssetVocabularyCategoriesSelector from './AssetVocabularyCategoriesSelector.es';
 
-/**
- * AssetCategoriesSelector is a component wrapping the existing Clay's MultiSelect component
- * that offers the user a tag selection input
- */
-class AssetCategoriesSelector extends Component {
-	_handleInputFocus(event) {
-		this.emit('inputFocus', event);
-	}
-
-	_handleSelectedItemsChange(event) {
-		this.emit('selectedItemsChange', event);
-	}
-}
+function AssetCategoriesSelector({
+	eventName,
+	groupIds,
+	id,
+	inputName,
+	onVocabulariesChange = () => {},
+	portletURL,
+	useFallbackInput,
+	vocabularies = []
+}) {
+	return (
+		<div id={id}>
+			{vocabularies.map((vocabulary, index) => {
+				const label = vocabulary.group
+					? `${vocabulary.title} ${vocabulary.group}`
+					: vocabulary.title;
 
-AssetCategoriesSelector.STATE = {
-	/**
-	 * Event name which fires when user selects a display page using item selector
-	 * @default undefined
-	 * @instance
-	 * @memberof AssetCategoriesSelector
-	 * @review
-	 * @type {?string}
-	 */
+				const isValid =
+					!vocabulary.required || vocabulary.selectedItems.length;
 
-	eventName: Config.string(),
+				return (
+					<AssetVocabularyCategoriesSelector
+						eventName={eventName}
+						groupIds={groupIds}
+						id={vocabulary.id}
+						inputName={inputName}
+						isValid={isValid}
+						key={index}
```

Comment:
In general, it's best to avoid numeric `key` values if you have an alternative that is more closely related to the "identity" of an element (and in this case, I think you do: I'm guessing the `vocabulary.id` would serve this purpose).

The reason is that React uses `key` as a hint to figure out how to best reconcile the DOM in a minimal way when changes are made. Imagine that you have a list of fruits:

1. Apple
2. Banana
3. Pear
4. Orange
5. Apricot
6. Peach

Now consider what happens if you filter the list so that "Orange" won't get rendered:

- If you use the names as keys, React can figure out that item "4. Orange" should be removed from the DOM, which is a single operation.
- If you use the numbers as keys, React thinks you're telling it that item 4's name changed from "Orange" to "Apricot", item 5's name changed from "Apricot" to "Peach", and item 6 was removed, which is three operations.

Reply:
:+1: 

Gotta get used to "thinking in React" but that's a great explanation thanks.
Will use the `vocabulary.id` as the `key`'s value.